### DISCUSSION TITLE
Example For Bulk Load

### DISCUSSION THREAD

**Discussion by gyansatapathy** [ben-manes/caffeine#588] [Q&A] [answered]
I would really appreciate if you can show how to override the loadAll in cache loader?

**Comment by ben-manes** [comment, marked as answer]
Hi @gyansatapathy,

You can simply implement the method and the cache will invoke it on a bulk request.

```java
Cache<UUID, User> users = Caffeine.newBuilder().build(new CacheLoader<UUID, User>() {
  public User load(UUID id) {
    return repository.getUserById(id);
  }
  public Map<UUID, User> loadAll(Set<UUID> ids) {
    return repository.getAllUsersById(ids);
  }
});

cache.get(Set.of(UUID.randomUUID(), UUID.randomUUID()));
```

You might also find the `CacheLoader.bulk` convenience function useful as a shorter syntax that delegates `load(key)` to the `loadAll(keys)` method.
```java
Cache<UUID, User> users = Caffeine.newBuilder()
    .build(CacheLoader.bulk((Set<UUID> keys) -> repository.getAllUsersById(ids));

cache.get(Set.of(UUID.randomUUID(), UUID.randomUUID()));
```

**Reply by gyansatapathy** [reply]
Thanks for the quick response. The issue is tried overriding both the loadAll and  load method . If I do cache.getAll(ids) it still calls the load method for every id in the list. Not sure what's wrong there.

I didn't try the second way you have mentioned here of only forcing the bulkLoad implementation. I will give it a shot.

**Reply by ben-manes** [reply]
An example in our test code is,

https://github.com/ben-manes/caffeine/blob/317e64708539a7b86fceab6c25592465f035354d/caffeine/src/test/java/com/github/benmanes/caffeine/cache/LoadingCacheTest.java#L182-L192

That provides a configured cache using a bulk loader defined as,
https://github.com/ben-manes/caffeine/blob/317e64708539a7b86fceab6c25592465f035354d/caffeine/src/test/java/com/github/benmanes/caffeine/cache/testing/CacheSpec.java#L498-L509

**Comment by gyansatapathy** [comment]
Hi @ben-manes I found out there was an issue in my classpath due to spring dependency management plugin as it was forcing an older version. Its working now. Thanks a ton.
